Add support to vulnerability aliases

macedogm opened the pull request:
This is an initial implementation that adds support to vulnerability aliases as documented in the [spec](https://github.com/openvex/spec/blob/52dd5645b4a37e723a27e7ac22f7e0e4abb5deed/OPENVEX-SPEC.md?plain=1#L333).

Note that support is only added when creating a new Vex document or adding a new statement. It doesn't support filtering or merging based on aliases.

Closes https://github.com/openvex/vexctl/issues/219

puerco reviewed (commented):
Nice! Thank you @macedogm 

Just one nit, can you add a simple validation to `vexStatementOptions.Validate()` ? I think for now checking we are not duplicating identifiers in `vexStatementOptions.Aliases` and `vexStatementOptions.Vulnerability` should be enough. (Oh and also add a test case to options_test.go )

https://github.com/openvex/vexctl/blob/8d81805748238ec2d0c8c95c266959ddee22ac19/internal/cmd/options.go#L72-L77

macedogm commented:
@puerco thanks for the quick review and comments. Yes, I'll add the improvements that you requested, just a note that I'll have to travel for a few days, so there might be some delay before I update the PR.

puerco commented:
@macedogm if you need this change we can handle the validation in a follow up. If you need this one quickly, just rebase the PR, we merge it and I can add the validation logic :)

macedogm commented:
@puerco I can wait a few days, no rush on that. It's more up to you if you prefer to merge this one quickly and do the rest on a follow up PR.

puerco commented:
Ah the linter is not happy, otherwise LGTM

macedogm commented:
@puerco hopefully I addressed your comments. PTAL and let me know if the style is as expected.

Additionally, should we consider adding a validation in `internal/cmd/add.go` to check if an alias being added is repeated with another aliases or vulnerabilities ID already present in the document? Would this validation make sense at the moment?

puerco commented:
The vexStatementOptions are embedded in the addOptions, so your change should work in both :) 

https://github.com/openvex/vexctl/blob/9bbf7cd70fc0d79f97ba21502aea40beec09f74d/internal/cmd/add.go#L36-L40


Thank you!
